glamod_marine_processing/qc_suite/modules/time_control.py

```
@@ -0,0 +1,676 @@
+            year2, month2, day2, hour2, year1, month1, day1, hour1
+        )
+
+    first_day = jul_day(year1, month1, day1) + hour1 / 24.0
+    last_day = jul_day(year2, month2, day2) + hour2 / 24.0
+
+    timdif = 24.0 * (last_day - first_day)
+
+    return timdif
+
+
+def last_month_was(year: int, month: int) -> (int, int):
+    """Short function to get the previous month given a particular month of interest
+
+    Parameters
+    ----------
+    year : int
+        Year of interest
+    month : int
+        Month of interest
+
+    Returns
+    -------
+    (int, int)
+        Year and month of previous month
+
+    """
+    last_year = year
+    last_month = month - 1
+    if last_month == 0:
+        last_month = 12
+        last_year = year - 1
+
+    return last_year, last_month
+
+
+def next_month_is(year: int, month: int) -> (int, int):
+    """Short function to get the next month given a particular month of interest
+
+    Parameters
+    ----------
+    year : int
+        Year of interest
+    month : int
+        Month of interest
+
+    Returns
+    -------
+    (int, int)
+        Year and month of next month
+    """
+    next_year = year
+    next_month = month + 1
+    if next_month > 12:
+        next_month = 1
+        next_year = year + 1
+
+    return next_year, next_month
+
+
+def year_month_gen(year1: int, month1: int, year2: int, month2: int) -> (int, int):
+    """A generator to loop one month at a time between year1 month1 and year2 month2
+
+    Parameters
+    ----------
+    year1 : int
+        Year of start month
+    month1 : int
+        Month of start month
+    year2 : int
+        Year of end month
+    month2 : int
+        Month of end month
+
+    Returns
+    -------
+    (int, int)
+        Return an iterator that yields tuples of a year and month
+    """
+    if year2 < year1:
+        return untestable
+    if not (0 < month1 <= 12):
+        return untestable
+    if not (0 < month2 <= 12):
+        return untestable
```

This should not return qc flags.

---

No, it should raise an error, I think.